include/deal.II/base/parameter_handler.h

```diff
@@ -125,6 +125,90 @@ namespace Patterns
      * assumption significantly, you can still overload this function.
      */
     virtual std::size_t memory_consumption () const;
+
+
+    /**
+     * Convert a compatible value type to a string. If the type is not
+     * compatible, an excepion ExcIncompatibleType is thrown.
+     *
+     * Internally, this function wraps the value in a boost::any object
+     * and calls the virtual function any_to_string(). Derived patterns
+     * should implement any_to_string for this function to work properly.
+     *
+     * An example usage is here:
+     *
+     * @code
+     * int a = 5;
+     * Patterns::Integer p;
+     *
+     * std::string s = p.to_string(a); // s = "5"
+     * @endcode
+     */
+    template<class T>
+    std::string to_string(const T &value) const;
+
+
+    /**
+     * Translate the content of the string matching this pattern, into a
+     * value and store it in @p value.  If the type T is not
+     * compatible with the pattern, an excepion ExcIncompatibleType is
+     * thrown.
+     *
+     * Internally, this function wraps a pointer to @p value in a
+     * boost::any object and calls the virtual function string_to_any(). Derived
+     * patterns should implement string_to_any for this function to work properly.
+     *
+     * An example usage is here:
+     *
+     * @code
+     * int a;
+     * Patterns::Integer p;
+     *
+     * p.to_value("5", a);
+     * @endcode
+     */
+    template<class T>
+    void to_value(const std::string &s, T &value) const;
+
+
+    /**
+     * An incompatible type was encountered when trying to access
+     * a pointer contained in a boost::any object
+     *
+     * @ingroup Exceptions
+     */
+    DeclException2 (ExcIncompatibleType,
+                    boost::any &, PatternBase &,
+                    << "The type " << arg1.type().name()
+                    << " is not compatible with the pattern "
+                    << arg2.description(Text));
+
+  protected:
+    /**
+     * Convert a string to a value, and store its content in the type
+     * pointed to by a boost::any object.
+     *
+     * Derived patterns can use this function to convert a matching string
+     * to a compatible type, stored as a boost::any object.
+     *
+     * This class expects the boost::any object to store a pointer to a
+     * value type that is compatible with this pattern. If this is not the case,
+     * an exception ExcIncompatibleType is thrown.
+     */
+    virtual void string_to_any(const std::string &s, boost::any &v) const;
```

Comment:
Because it needs to inspect the v that is passed to it, to verify if is compatible with the pattern. How would you choose between `int` and `unsigned int` for `Patterns::Integer`, for example?